bluesky/utils.py

```diff
@@ -183,45 +189,40 @@ def __init__(self, RE):
         self.num_sigints_processed = 0  # count SIGINTs processed
 
     def __enter__(self):
-        self.RE.loop.call_later(0.1, self.check_for_signals)
         return super().__enter__()
 
-    def check_for_signals(self):
+    def handle_signals(self):
         # Check for pause requests from keyboard.
+        # TODO, there is a possible race condition between the two
```

Comment:
For future reference, I had pasted this into our slack chat to explain multiple signals being ignored:
https://www.gnu.org/software/libc/manual/html_node/Checking-for-Pending-Signals.html
`Remember that if there is a particular signal pending for your process, additional signals of that same type that arrive in the meantime might be discarded. For example, if a SIGINT signal is pending when another SIGINT signal arrives, your program will probably only see one of them when you unblock this signal.`

Reply:
added do docstring of base class, I think the race condition here is that the defer and hard pause tasks may both run at the same time.